v2/client/invoker.go

```diff
@@ -61,14 +64,22 @@ func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn p
 				cecontext.LoggerFrom(ctx).Error(fmt.Errorf("cloudevent validation failed on response event: %v, %w", verr, err))
 			}
 		}
-		if respFn != nil {
-			var rm binding.Message
-			if resp != nil {
-				rm = (*binding.EventMessage)(resp)
+
+		// protocol can manual ack by the result
+		if respFn == nil {
+			if result != nil {
+				err = m.Finish(result)
```

Comment:
result could be non-nil and an ACK, and respFn could return back a NACK and will have done the wrong thing here.

Reply:
ok, this can be  `protocol.IsACK(result)`, it should more safe